python/triton/compiler/code_generator.py

```diff
@@ -1349,6 +1349,9 @@ def visit_Attribute(self, node):
         lhs = self.visit(node.value)
         if _is_triton_tensor(lhs) and node.attr == "T":
             return semantic.permute(lhs, (1, 0), builder=self.builder)
+        # NOTE: special case ".value" for BC
+        if isinstance(lhs, constexpr) and node.attr != "value":
+            lhs = lhs.value
```

Comment:
A little bonus, this automatically unwraps constexpr values so you never need to explicitly write `x.value` for constexpr types.